templates/site/js/main.js

```diff
@@ -114,129 +49,138 @@ function login(){
-    let authTkt;
-    var json = response.json();
-
-    try{
-
-        if(json.code == 200){
+function getSessionDetails(){
+    const sessionURLFragment = "/magpie/session";
 
-            var cookieArray = response.headers.getSetCookie();
+    return fetch(sessionURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => { return response.json().then(json =>{
+                return json;
+            })
+        })
+}
 
-            if(cookieArray.length > 0){
-                authTkt = cookieArray[0];
-                return authTkt;
+//User functions
+//--------------
+function checkUserAuthenticated(){
+    getSessionDetails().then(json => {
+        if(json.code && json.code == 200) {
+            if (json.authenticated === false) {
+                window.location.href = loginHome;
             }
         }
-        else{
-            return null;
-        }
-
-    }
-    catch (error){
-         //throw new Error(error)
-        console.log("Error caught:")
-        console.log(error);
-
-    }
+    })
 }
 
-function getNodeRegistry() {
-    const githubURL = "{{ node_registry_url }}";
-    return fetch(githubURL).then(response => {
-        return response.json().then(json => {
+function getUserDetails(){
+    const loggedUserURLFragment = "/magpie/users/current";
+    return fetch(loggedUserURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => {return response.json()}).then(json =>{
             return json;
         })
-    })
 }
-function getBaseURL(json){
-    return json.then( sessionData =>
-    {
-        var urlArray = sessionData.url.split("/magpie");
-        return urlArray[0];
+
+function updateUserDetails(){
+    var username;
+    var sessionEmail;
+
+    getUserDetails().then(json => {
+        username = json.user["user_name"];
+        sessionEmail = json.user["email"];
+        var password = document.getElementById("userPassword").value;
+        var email = document.getElementById("settingsEditEmail").value;
+
+        showSpinner()
+
+        if (password != "" || password != null){
+            updateUserPassword(username, password);
+        }
+
+        if(email != sessionEmail && email != null){
+            updateUserEmail(username, email);
+        }
     })
+
 }
 
-function getSessionDetails(){
-    //TODO Remove sessionURL and use sessionURLFragment
-    //TODO Remove nodeURL and use {MAGPIE_URL}
-    const sessionURLFragment = "session";
-    const sessionURL = nodeURL + "/magpie/session"
-    //const testSessionURL = testNodeURL + "/magpie/session"
-    return fetch(sessionURL,{
-        method: "GET",
+function updateUserEmail(username,email){
+    const updateURLFragment = "/magpie/users/" + username;
```

Comment:
It isn't necessary to get the current user here. Just use the `/magpie/users/current` path

Reply:
Removed this function.